base/graveyard.hpp

```diff
@@ -0,0 +1,26 @@
+
+#pragma once
+
+#include <cstdint>
+#include <memory>
+
+#include "base/thread_pool.hpp"
+
+namespace principia {
+namespace base {
+
+template<typename T>
+class Graveyard {
```

Comment:
Templatize `Bury`, not the class, that way a single `Graveyard` can be used for different types.

Reply:
Done.